Sources/StreamVideoSwiftUI/Utils/VideoRendererPool/VideoRendererPool.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import Foundation
+import StreamVideo
+
+/// A pool for managing reusable `VideoRenderer` instances.
+final class VideoRendererPool {
+    /// The underlying pool of `VideoRenderer` instances managed by this pool.
+    private let pool: ReusePool<VideoRenderer>
+    /// A cancellable object to observe call end notifications for releasing all renderers.
+    private var callEndedCancellable: AnyCancellable?
+
+    /// Initializes the `VideoRendererPool` with a specified initial capacity.
+    ///
+    /// - Parameter initialCapacity: The initial capacity of the pool (default is 10).
+    init(initialCapacity: Int = 10) {
+        // Initialize the pool with a capacity and a factory closure to create `VideoRenderer` instances
+        pool = ReusePool(initialCapacity: initialCapacity) {
+            VideoRenderer(frame: CGRect(origin: .zero, size: .zero))
```

Comment:
One downside is that we always create 10 views, even for direct calls.

Reply:
that's ok for 2 reasons: 1) VideoRenderers that don't have a track assigned on them are just plain views that do nothing 2) we can very easily make this configurable and even dynamically changed based on the active call (e.g. how many members are set in the Call that's the number of the videoRenderers we will create)(one more point here is that going lower won't hurt us as we the pool can expant to accomodate the call's needs)

Reply:
No, I think it's fine like this. 10 UIViews won't impact the memory that much.